File: python/461_Hamming_Distance.py
'''
First key point: XOR.
Second key point: z &= (z-1) will knock out the lowest one. 
'''

class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        z = x ^ y
        count = 0
        while (z):
            count += 1;
            z &= (z-1)

        return count

File: javascript/461_Hamming_Distance.js
/*
First key point: XOR.
Second key point: z &= (z-1) will knock out the lowest one.
*/
/**
 * @param {number} x
 * @param {number} y
 * @return {number}
 */
var hammingDistance = function(x, y) {
    let z = x ^ y;
    let count = 0;
    while (z) {
        count++;
        z &= z - 1;
    }

    return count;
};
